src/praisonai/praisonai/agents_generator.py

```
@@ -701,6 +701,148 @@ def generate_crew_and_kickoff(self):
+                            continue
+                        for t in task_cfg.get('tools') or []:
+                            if isinstance(t, str) and t.strip():
+                                needed_tools.add(t.strip())
+
+                # Resolve only the tools actually referenced in YAML
+                for tool_name in needed_tools:
+                    try:
+                        tool_instance = self._get_tool_by_name(tool_name)
+                        if tool_instance:
+                            tools_dict[tool_name] = tool_instance
+                    except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)